src/popup/components/TransactionTagList.vue

```diff
@@ -0,0 +1,157 @@
+<template>
+  <div
+    class="transaction-tag-list"
+    :class="{ dense }"
+  >
+    <TransactionTag
+      v-for="label in labels"
+      :key="label"
+      :tx-type="label"
+      class="title-tag"
+      data-cy="label"
+    />
+  </div>
+</template>
+
+<script lang="ts">
+import { computed, defineComponent, PropType } from '@vue/composition-api';
+import { SCHEMA } from '@aeternity/aepp-sdk';
+import { TranslateResult } from 'vue-i18n';
+import { useTransactionTx } from '../../composables';
+import { useGetter, useState } from '../../composables/vuex';
+import {
+  IAccount,
+  INetwork,
+  ITokenList,
+  ITx,
+  TransactionType,
+  TxFunctionRaw,
+} from '../../types';
+import { i18n } from '../../store/plugins/languages';
+import {
+  AENS,
+  DEX,
+  FUNCTION_TYPE_DEX,
+  TX_FUNCTIONS,
+  TX_TYPE_MDW,
+} from '../utils';
+
+import TransactionTag from './TransactionTag.vue';
+
+export default defineComponent({
+  components: { TransactionTag },
+  props: {
+    tx: { type: Object as PropType<ITx>, default: null },
+    isIncomplete: Boolean,
+    isPending: Boolean,
+    isClaim: Boolean,
+    dense: Boolean,
+  },
+  setup(props, { root }) {
+    const { tx } = props;
+
+    const {
+      txType,
+      isAllowance,
+      isDex,
+    } = useTransactionTx({ store: root.$store, tx });
+
+    const account = useGetter<IAccount>('account');
+    const activeNetwork = useGetter<INetwork>('activeNetwork');
+    const availableTokens = useState<ITokenList>('fungibleTokens', 'availableTokens');
+    const getTxDirection = useGetter('getTxDirection');
+
+    const labels = computed<(string | TranslateResult)[]>(() => {
+      if (!tx) return [];
+
+      const transactionTypes = i18n.t('transaction.type') as Record<TransactionType, TranslateResult>;
```

Comment:
The types here are wrong. The `transaction.type` translation keys are actually camelCase: "spendTx", "contractCreateTx", "contractCallTx"... and the `TransactionType` are PascalCase: "SpendTx", "ContractCreateTx", "ContractCallTx".